doc/paikannuspohjan_kontekstit.md

```diff
@@ -0,0 +1,321 @@
+joukot suunniteltuja muutoksia. Käyttäjän kannalta ne toimivat pitkälti kuten luonnos paikannuspohja, mutta ne
+ovat kukin omia erillisiä kontekstejaan.
+
+Suunnitelmille tehdään omia julkaisuja (toisin kuin luonnokselle, jonka julkaisu on virallisen paikannuspohjan muutos).
+Julkaistu suunnitelma viedään Ratkoon suunniltuina käsitteinä, jolloin niihin voidaan suunnitella myös muita kohteita
+Ratkon puolella.
+
+Suunniteltu paikannuspohja rakentuu aina virallisen paikannuspohjan päälle, eikä siis ole mitenkään tietoinen luonnos
+paikannuspohjasta. Kun suunnitelman käsitteet kuitenkin valmistuvat, ne viedään ensin luonnoksiksi. Luonnospuolella
+niitä voidaan edelleen muokata ja täydentää ennen kun ne julkaistaan normaalin luonnoksen tapaan viralliseksi. Tässä
+kohtaa myös suunnitelmajulkaisu päivitetään valmistuneeksi.
+
+## ID käsittely ja viittaukset kontekstien välillä
+
+Eri kontekstien versiot samasta käsitteestä ovat kannassa omia rivejään ja siten niillä on omat ID:nsä. Käyttöliittymän
+kannalta käsittellä on kuitenkin aina vain yksi ID ja lisätietona konteksti jossa sitä tarkastellaan. Tuo yksi ID on
+käsitteenä nimeltään virallinen ID (official ID). Käytännössä se on käsitteen ensimmäisen ilmenemän ID, riippumatta
+missä kontekstissa se ensimmäisen kerran syntyi. Koska tuo ID ei muutu käsitteen siirtyessä eri kontekstien välillä,
+käyttöliittymä voi helposti siirtyä tarkastelemaan koko paikannuspohjaa eri konteksteissa.
+
+Myös käsitteiden keskinäiset viittaukset tehdään aina tällä virallisella ID:llä, minkä johdosta noita viittauksia ei
+tarvitse korjata silloin kun käsiteestä tehdään uusi versio toiseen kontekstiin -- virallinen ID säilyy aina samana.
+Koska luonnos ja suunnitelmat rakentuvat virallisen paikannuspohjan päälle, tämä tarkoittaa että niiden sisältämät
+käsitteet voivat viitata virallisilla ID:llä sekä oman kontekstin käsitteisiin että virallisiin käsitteisiin.
+
+Lopullinen rivi jota viite tarkoittaa pitää siis aina tulkita huomioiden myös konteksti. Jos esimerkiksi viralliseen
+käsitteeseen tulee muutos luonnoskontekstiin, viite olioon säilyy muuttumattomana ja osoittaa nyt luonnosversioon.
+Vastaavasti jos käsite poistuu luonnoskontekstista, viite tarkoittaa nyt virallisesta paikannuspohjasta tulevaa oliota.
+
+Käsitteiden elinkaaria uuden luonnissa tai olemassaolevan muutoksissa kuvataan alla tarkemmin eri skenaarioille
+kontekstit huomioiden. Huomattavaa on, että ensimmäisenä syntynyt olio (tietokantarivi) säilyy ja siirtyy lopulta
+viralliseen paikannuspohjaan, mutta muokattaessa muutokset tehdään kopioon, josta data kopioidaan alkuperäiselle
+julkaisussa. Tämä varmistaa että virallinen ID säilyy muita viittaajia varten kaikissa ketjuissa.
+
+### Kaavioiden merkinnät
+
+```mermaid
+graph TD
+    subgraph legendColor [Värikoodit]
+        legendDraft(Luonnos olio)
+        legendOfficial(Virallinen olio)
+        legendPlan(Suunniteltu olio)
+        legendDraft ~~~ legendOfficial ~~~ legendPlan
```

Comment:
Näkymättömät viivat ovat mermaidissa ihan dokumentoitu tapa hallita (hiukan) layouttia. Ilman tätä, nuo osat piirtyy vierekkäin ja diagrammi menee leveäksi.